src/core/qgsofflineediting.cpp

```diff
@@ -528,31 +528,30 @@ QgsVectorLayer *QgsOfflineEditing::copyVectorLayer( QgsVectorLayer *layer, sqlit
   // add geometry column
   if ( layer->isSpatial() )
   {
-    QString geomType = QLatin1String( "" );
-    switch ( layer->wkbType() )
+    QString geomType = wkbTypeAsString( layer->wkbType() );
+    // We cannot trust WFS wkbType because it may be deduced by
+    // the first feature from a GetFeature request, but a dataset
+    // can be (at least in QGIS server) composed by features with
+    // simple and MULTI geometries of the same type, for example
+    // POLYGON and MULTIPOLYGON is a quite common scenario.
+    if ( ! geomType.contains( QStringLiteral( "MULTI" ) ) && layer->providerType() == QLatin1Literal( "WFS" ) )
```

Comment:
Note: the same can happen with a shapefile layer

Reply:
This should be using `QgsWkbTypes::isMultiType` instead of string comparison.